[DEVX-4943] Terminus 4 documentation and release note. (#9546)
pantheon-systems/documentation: source/releasenotes/2025-05-07-terminus-4-0-0.md

---
title: "Terminus 4.0.0 release now available"
published_date: "2025-05-07"
categories: [tools-apis]
---

Terminus [4.0.0](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0) is now available.

> The next generation of Terminus adds support for PHP 8.4 and removes support for PHP versions 8.1 and below.

Reviewer: Is it worth mentioning how this relates to Pantheon sites running PHP 7.x and below?

Reviewer 2: Since terminus runs on local machines, the version of PHP that's running on Pantheon application servers seems less relevant (and probably confuses things)

Reviewer 3: I think we should explain that

> The next generation of Terminus adds support for PHP 8.4 and removes support for PHP versions 8.1 and below.

Reviewer 3: ```suggestion
This release adds support for PHP 8.4 and removes support for PHP versions 8.1 and below.
```

We're about to use the word "generation" to mean something different as we do a more complicated and intertwined update to container infra. We should not use it here when we mean something much simpler.

Reviewer 2: ```suggestion
This release adds support for PHP 8.4 and removes support for PHP versions 8.1 and below.
Terminus 4 is compatible with PHP 8.2 and 8.3.
See [Supported Terminus and PHP versions](/terminus/supported-terminus).
The PHP version refers to the local machine on which Terminus itself runs (e.g. developers' own computers or containers for continuous integration). This is distinct from PHP versions that run websites on Pantheon's infrastructure (which still include support for PHP 7.x).
```

For more information about this release, visit the [GitHub release page](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0).

> ## Support for older PHP versions in Terminus 3.x
> 
> Terminus 3.x is **not compatible with PHP 8.4**. The recommended PHP version to use locally with Terminus 3.x is 8.3. 

Reviewer 3: ```suggestion
## Terminus 3 to be supported through May 7th, 2026

It is Pantheon's policy to support a version of Terminus for one year from the release of the subsequent version. With the release of Terminus 4 today, Pantheon will continue to support Terminus 3 (bug fixes, but not new feature development) through May 7th, 2026.
```

As written, this heading sets the expectation that Terminus 4 is not compatible with PHP 8.3 (I know it doesn't say that but @jazzsequence and I have found ourselves confused a few times). We've made the compability stuff clearer above and elsewhere. Somewhat relatedly, we should highlight the EOL date for Terminus 3.

## How to upgrade to Terminus 4.x

If you previously used Homebrew (macOS-only) to manage your installation, you could upgrade using:

```shell{promptUser: user}
brew upgrade terminus
```

> If you installed directly from the `.phar` file, you could upgrade using `self:update` command:

Reviewer 3: ```suggestion
If you installed Terminus directly from the `.phar` file, you should upgrade using `self:update` command:
```

---
title: "Terminus 4.0.0 release now available"
published_date: "2025-05-07"
categories: [tools-apis]
---

Terminus [4.0.0](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0) is now available.
This release adds support for PHP 8.4 and removes support for PHP versions 8.1 and below.
Terminus 4 is compatible with PHP 8.2 and 8.3.
See [Supported Terminus and PHP versions](/terminus/supported-terminus).
The PHP version refers to the local machine on which Terminus itself runs (e.g. developers' own computers or containers for continuous integration). This is distinct from PHP versions that run websites on Pantheon's infrastructure (which still include support for PHP 7.x).

For more information about this release, visit the [GitHub release page](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0).

## Terminus 3 to be supported through May 7th, 2026

> It is Pantheon's policy to support a version of Terminus for one year from the release of the subsequent version. With the release of Terminus 4 today, Pantheon will continue to support Terminus 3 (bug fixes, but not new feature development) through May 7th, 2026.

Reviewer 4: This doesn't quite describe the policy. We allow folks to *RUN* Terminus 3.6.2 until May 07, 2026. This doesn't necessarily mean that we will also fix bugs in this release, though, as that is a separate policy. Supporting bug fixes in Terminus 3 is at the discretion of @scottbuscemi .

Reviewer 2: I've removed this section. It doesn't make sense to include if the EoL date is not representative of support.

> ## Terminus 3 to be supported through May 7th, 2026
> 
> It is Pantheon's policy to support a version of Terminus for one year from the release of the subsequent version. With the release of Terminus 4 today, Pantheon will continue to support Terminus 3 (bug fixes, but not new feature development) through May 7th, 2026.

Reviewer 2: ```suggestion
```
